src/test/java/it/geosolutions/httpproxy/HttpProxyTest.java

```diff
@@ -20,96 +20,261 @@
 package it.geosolutions.httpproxy;
 
 import java.io.File;
+import java.io.IOException;
 import java.net.URL;
 import java.net.URLEncoder;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
 
 import javax.servlet.ServletConfig;
 import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletInputStream;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import org.apache.commons.httpclient.Header;
-import org.apache.commons.httpclient.HttpClient;
-import org.apache.commons.httpclient.methods.GetMethod;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import it.geosolutions.httpproxy.jetty.Start;
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.entity.InputStreamEntity;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicHeader;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.mockito.Matchers;
 import org.mockito.Mockito;
 
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
```

Comment:
What we are missing here is the tests for the new properties (that trigger this update of the library), basically, we will need a "network-proxy" to validate this and set these properties during the test (example values):
```
-Dhttp.proxyHost=proxyhttp.comune.intranet 
-Dhttp.proxyPort=8080 
-Dhttps.proxyHost=proxyhttps.comune.intranet 
-Dhttps.proxyPort=8080 
-Dhttp.nonProxyHosts='localhost|*.comune.intranet|192.168.*'
```

The hard bit here is how to simulate a network proxy for our tests?

My suggestion would be to rely on [WireMock](http://wiremock.org/docs/proxying/) and create three mocked servers:
- The first will act as the network proxy, as described [HERE](http://wiremock.org/docs/proxying/).
- The second (like you are using now) will act as the remote HTTP server we want to reach through the network-proxy.
- The third will act as the internal HTTP server that we don't want to reach via the network-proxy (-Dhttp.nonProxyHosts).

What I'm not sure yet, is out to validate if the requested was or was not routed through the proxy, can you check that @shadab-geo ?